src/cobertura.d

```diff
@@ -0,0 +1,199 @@
+		}
+	}
+}
+
+enum isNullable(T) = is(T : Nullable!P, P);
+
+auto filterNullable(Range)(Range r) if (isNullable!(ElementType!Range)) {
+	import std.algorithm : filter, map;
+	return r.filter!(a => !a.isNull).map!(a => a.get);
+}
+
+private template andThen(alias fun) {
+    auto andThen(T)(Nullable!T t) {
+        alias RT = typeof(fun(T.init));
+        static if (is(RT == void)) {
+            if (!t.isNull)
+                fun(t.get);
+        } else {
+            alias Result = Nullable!RT;
+            if (t.isNull)
+                return Result.init;
+            return Result(fun(t.get));
+        }
+    }
+}
+
+FileCoverage parseLstFile(Lines)(Lines r, string path) if (isInputRange!Lines &&
+													   isSomeString!(ElementType!Lines)) {
+	import std.range : enumerate, tee;
+	import std.algorithm : map, stripLeft, until;
+	import std.array : array;
+	size_t linesCovered = 0;
+	auto lines = r.enumerate.map!((line){
+			return line
+				.value
+				.stripLeft(' ')
+				.until('|')
+				.tryTo!size_t
+				.andThen!(v => LineCoverage(line.index, v));
+		})
+		.filterNullable
+		.tee!(line => linesCovered += (line.count > 0))
+		.array();
+	return FileCoverage(path, lines, linesCovered);
+ }
+
+unittest {
+	import unit_threaded;
+	import std.string;
+	testLst
+		.splitLines
+		.parseLstFile("file").should == FileCoverage("file", [LineCoverage(9, 1), LineCoverage(13, 0), LineCoverage(14, 0), LineCoverage(15, 0), LineCoverage(19, 0)], 1);
+}
+
+auto generateXmlFile(FileCoverage[] files) {
+	import core.stdc.time: time;
+	import std.format: format;
+	import std.path: buildPath;
+
+	import std.algorithm.iteration: splitter, map, sum;
+
+	size_t linesValid = files.map!(file => file.lines.length).sum();
+	size_t linesCovered = files.map!(file => file.linesCovered).sum();
+
+	double lineRate = cast(double)linesCovered / linesValid;
+
+	string res = `<?xml version="1.0"?>
+<coverage version="5.3"
+	timestamp="%s"
+	lines-valid="%s"
+	lines-covered="%s"
+	line-rate="%s"
+	branches-covered="0"
+	branches-valid="0"
+	branch-rate="0"
+	complexity="0"
+>
+`.format(time(null), linesValid, linesCovered, lineRate);
+
+	res ~= "\t<sources>\n\t\t<source>./</source>\n\t</sources>\n";
+	res ~=
+		`	<packages>
+`;
+	foreach(file; files) {
+		lineRate = cast(double)file.linesCovered / file.lines.length;
+
+		string fpath = buildPath(file.path[0 .. $-4].splitter('-')) ~ ".d";
+
+		res ~=
+			`		<package name="covered" line-rate="%1$s" branch-rate="0" complexity="0">
+			<classes>
+				<class name="%2$s" filename="%2$s" complexity="0" line-rate="%1$s" branch-rate="0"><methods></methods>
+`.format(lineRate, fpath);
+
+		res ~= "\t\t\t\t\t<lines>\n";
+		foreach(line; file.lines) {
+			res ~= "\t\t\t\t\t\t<line number=\"%s\" hits=\"%s\"/>\n".format(line.index+1, line.count);
+		}
+		res ~= "\t\t\t\t\t</lines>\n\t\t\t\t</class>\n\t\t\t</classes>\n\t\t</package>\n\t";
+	}
+	res ~= "</packages>\n</coverage>";
+	return res;
+}
+
+unittest {
+	import unit_threaded;
```

Comment:
pulling in this package for one call to should is overkill IMO

Reply:
There are two calls actually, but you have a point.

That said, I never leave the house without unit_threaded